supportedBackends/local/src/main/scala/cromwell/backend/impl/local/LocalBackend.scala

```diff
@@ -1,51 +1,229 @@
-  val fileSystems = List(FileSystems.getDefault)
-  val workflowPaths = new WorkflowPaths(jobDescriptor.descriptor, backendConfiguration, fileSystems)
+  import LocalBackend._
+  import better.files._
+  import cromwell.core.PathFactory._
+
+  import scala.sys.process._
+
+  // Mutable process variable assigned when the execute method is called.
+  // Needs to be accessible to be killed when the job is aborted.
+  private var process: Option[Process] = None
+
+  val workflowDescriptor = jobDescriptor.descriptor
+  val jobPaths = new JobPaths(workflowDescriptor, backendConfiguration, jobDescriptor.key)
+  val fileSystemsConfig = backendConfiguration.getConfig("filesystems")
+  override val sharedFsConfig = fileSystemsConfig.getConfig("local")
+
+  val call = jobDescriptor.key.call
+  val callEngineFunction = {
+    val callContext = new CallContext(
+      jobPaths.callRoot.toString,
+      jobPaths.stdout.toAbsolutePath.toString,
+      jobPaths.stderr.toAbsolutePath.toString
+    )
+
+    new LocalCallEngineFunctions(fileSystems, callContext)
+  }
+
+  val evaluator = {
+    val declarations = workflowDescriptor.workflowNamespace.workflow.declarations ++ call.task.declarations
+    val unqualifiedWorkflowInputs = workflowDescriptor.inputs map {
+      case (fqn, v) => splitFqn(fqn)._2 -> v
+    }
+    val knownInputs = unqualifiedWorkflowInputs ++ jobDescriptor.symbolMap
+    val lookup = WdlExpression.standardLookupFunction(knownInputs, declarations, callEngineFunction)
+    new ValueEvaluator(lookup, callEngineFunction)
+  }
+
+  private def evaluate(wdlExpression: WdlExpression) = evaluator.evaluate(wdlExpression.ast)
+
+  // TODO determine how this plays with the init/validation work
+  val runtimeAttributes = {
+    val evaluateAttrs = call.task.runtimeAttributes.attrs mapValues evaluate
+    // Fail the call if runtime attributes can't be evaluated
+    val evaluatedAttributes = TryUtils.sequenceMap(evaluateAttrs, "Runtime attributes evaluation").get
+    LocalRuntimeAttributes(evaluatedAttributes)
+  }
+
+  lazy val runsOnDocker = runtimeAttributes.dockerImage.isDefined
+  lazy val processArgs = {
+    val dockerRun = runtimeAttributes.dockerImage map buildDockerRunCommand getOrElse ""
+    Command(Seq("/bin/bash", "-c", s"cat ${jobPaths.script} | $dockerRun /bin/bash <&0"))
+  }
+
+  // Stream Writers
+  lazy val stdoutWriter = jobPaths.stdout.untailed
+  lazy val stderrTailed = jobPaths.stderr.tailed(100)
 
   override def preStart() = {
-    workflowPaths.workflowRoot.createDirectories()
+    // workflowPaths.workflowRoot.createDirectories() TODO move to initialize actor
+    jobPaths.callRoot.createDirectories()
   }
-   /**
+
+  lazy val executionPromise = Promise[BackendJobExecutionResponse]()
+
+  // We don't want the job to run on the same execution context as the actor, because it would block the actor message processing abilities while the job is running
+  override implicit val ec = scala.concurrent.ExecutionContext.global
```

Comment:
I don't think it's worth doing at this moment but could you add a ticket to get this off of the global context?